app/controllers/api/v1/users_controller.rb

```diff
@@ -7,7 +7,7 @@ class UsersController < ApiController
 
       skip_before_action :ensure_authenticated, only: %i[create]
 
-      before_action only: %i[show update destroy purge_avatar] do
+      before_action except: %i[create change_password] do
```

Comment:
Whitelisting vs blacklisting strategies like with authentication authorization,
With the old approach adding a new API will risk exposing it unintentionally,
Now any API that is not explicitly declared will require the highest permissions to pass and can be spotted easily without introducing a security vulnerability,
If you're okay with this we may want to propagate it through the code base and make a de-facto,

Reply:
I'm not following - isn't whitelisting more secure?

Reply:
Yes, whitelisting is more secure.
The old behavior conceptually was a blocklist (all are unrestricted except those mentioned explicitly in the `:only` list).
Now it's more of a whitelist where by default any API in the users controller will be restricted unless those mentioned explicitly in the whitelist (the `except:` list).